frontend/src/components/common/Admin/RemoveUserModal.tsx

```diff
@@ -1,33 +1,76 @@
+import { gql, useMutation, useQuery } from "@apollo/client";
 import {
   Button,
+  Divider,
   Modal,
   ModalContent,
   ModalFooter,
   ModalOverlay,
   useDisclosure,
 } from "@chakra-ui/react";
-import React from "react";
+import React, { useContext } from "react";
+import AuthContext from "../../../contexts/AuthContext";
 import { CloseOutlineIcon } from "../icons";
-import ModalText from "../ModalText";
+import ModalText from "./ModalText";
 import RemoveUserConfirmationModal from "./RemoveUserConfirmationModal";
+import RemoveUserErrorModal from "./RemoveUserErrorModal";
+import { Role } from "../../../types/AuthTypes";
 
 interface RemoveUserModalProps {
   name: string;
   email: string;
   onCloseParent: () => void;
 }
 
+interface UserGQLObject {
+  userByEmail: { id: string };
+}
+
+const GET_USER = gql`
+  query RemoveUserModal_UserByEmail($email: String!) {
+    userByEmail(email: $email) {
+      id
+    }
+  }
+`;
+
+const REMOVE_USER = gql`
+  mutation RemoveUserModal_DeleteUserByEmail($email: String!) {
```

Comment:
maybe we can just call this `DeleteUser` for now. I think we're going to refactor this to the `APIClients` folder down the line for general use, so we can leave it more generic

Reply:
should i make that refactor in this PR?

Reply:
no, I think we can make it a separate change, to keep this one modular